### PR TITLE
Tweak z-index

### PR DESCRIPTION
**Pull request by amanmahajan7**
Fixes row selection styles

image

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -1236,10 +1220,36 @@ function DataGrid<R, SR, K extends Key>(
             </>
           )}
         </RowSelectionChangeProvider>
-
-        {/* render empty cells that span only 1 column so we can safely measure column widths, regardless of colSpan */}
-        {renderMeasuringCells(viewportColumns)}
       </DataGridDefaultRenderersProvider>
+
+      {/* render empty cells that span only 1 column so we can safely measure column widths, regardless of colSpan */}
+      {renderMeasuringCells(viewportColumns)}
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:1226]: Moved it outside the provider.

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -1236,10 +1220,36 @@ function DataGrid<R, SR, K extends Key>(
...
+
+      {/* extra div is needed for row navigation in a treegrid */}
+      {hasGroups && (
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:1229]: Move this div to the end. This way we can use a lower value of `zIndex` and it already uses `pointer-events: none`. Can you think of any issues with this change?

File: src/DataGrid.tsx [hunk 3]
```diff
@@ -1236,10 +1220,36 @@ function DataGrid<R, SR, K extends Key>(
...
+        <div
+          ref={rowRef}
```
Comment (nstepien) [inline src/DataGrid.tsx:1231]: Should re rename `rowRef`?

File: src/HeaderRow.tsx [hunk 4]
```diff
@@ -32,14 +32,14 @@ const headerRow = css`
     font-weight: bold;
 
     & > .${cell} {
-      /* Should have a higher value than 0 to show up above regular cells */
-      z-index: 1;
+      /* Should have a higher value than 1 to show up above regular cells */
```
Comment (nstepien) [inline src/HeaderRow.tsx:35]:
```suggestion
      /* Should have a higher value than 1 to show up above regular cells and the focus sink */
```

**Review by nstepien — DISMISSED** [review]

**Review by nstepien — APPROVED** [review]

**Pull request merged**